File: BlogApplication/src/main/java/com/example/demo/Service/PostService.java
package com.example.demo.Service;

import java.util.List;

import com.example.demo.Entity.Post;

public interface PostService {
	Post savePost(Post post, Long userId);
	
    List<Post> getAllPosts();
    
    Post getPostById(Long id);
    
    void likePost(Long postId);
    
    List<Post> searchByName(String name);
    
    List<Post> getPopularPost();
    
    List<Post> getPostsByUserId(Long userId);
    
    void deletePostById(Long postId);

	Post updatePost(Post post);
}

File: blogweb/blogweb/src/app/service/post.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { UserService } from './user.service';

const BASIC_URL = 'http://localhost:8080';

@Injectable({
  providedIn: 'root'
})
export class PostService {


  constructor(private http:HttpClient) { }

  createNewPost(data:any):Observable<any>{
    return this.http.post(BASIC_URL+ `/api/posts/${data.userId}`, data);
  }

  getAllPosts():Observable<any>{
    return this.http.get(BASIC_URL+ '/api/posts');
  }

  getPostById(postId:number):Observable<any>{
    return this.http.get(BASIC_URL+ `/api/posts/${postId}`);
  }

  likePost(postId:number):Observable<any>{
    return this.http.put(BASIC_URL+ `/api/posts/${postId}/like`,{});
  }

  searchByName(name:string):Observable<any>{
    return this.http.get(BASIC_URL+ `/api/posts/search/${name}`);
  }

  getPopularPost():Observable<any>{
    return this.http.get(BASIC_URL+ '/api/posts/popular');
  }

  updatePost(post: any):Observable<any> {
    console.log(post);
    
    return this.http.put(BASIC_URL+ `/api/posts/${post.id}`,post);
  }

  getPostByUserId(userId:number){
    return this.http.get(BASIC_URL+ '/api/posts/user/'+userId);
  }

  deletePost(id:number){
    return this.http.delete(BASIC_URL+'/api/posts/'+id);
  }
}
